src/ui/pages/Contract.tsx

```diff
@@ -50,7 +52,7 @@ export function Contract() {
 
   const [tabIndex, setTabIndex] = useState(TABS.findIndex(({ id }) => id === activeTab) || 1);
 
-  const [isOnChain, setIsOnChain] = useState(true);
+  const [isOnChain, setIsOnChain] = useState<boolean | null>(null);
```

Comment:
it either is on chain or is not on chain, why does this variable need to be null?

Reply:
null indicates the check has not been performed yet

Reply:
why do we need that? i think `true` is a sensible default. if it turns out that is not on-chain, then we display the "not on-chain" text, otherwise we display a reinstantiate button. keep in mind that the data for the page comes from local storage and URL, regardless of the contract being on chain or not.

Reply:
I noticed it was causing the page to load as though the contract were still present and then update to show that it wasn't. It's a better user experience in my opinion to just defer showing the final page until it's known one way or another.